### PR TITLE
add navigator clipboard, ClipboardItem

### PR DESCRIPTION
**Pull request by keithamus**
This implements `navigator.clipboard.write` and `navigator.clipboard.read`, which both work with `ClipboardItem`s. It also adds a re-implementation of `ClipboardItem` which supports Promises, adhering to the [spec](https://w3c.github.io/clipboard-apis).

[`write`, per the spec](https://w3c.github.io/clipboard-apis/#dom-clipboard-write) takes an array of `ClipboardItem`s and writes to the clipboard. We're limited in what we can do here, so we extract the `text/plain` content of the first one and call `writeText`. This is a pretty reasonable default behaviour.

[`read`, per the spec](https://w3c.github.io/clipboard-apis/#dom-clipboard-read) should return an array of `ClipboardItem`s representing the user's current clipboard. We're limited what we can do here, so we call `readText` and convert it into a `ClipboardItem` with just `text/plain`.

[`ClipboardItem`](https://w3c.github.io/clipboard-apis/#clipboard-item-interface) has a `presentationStyle` attribute (defaulting to `'unspecified'` unless it's passed as an option), and `getType()` always returns `Blob`s. In Chrome's buggy implementation, `getType('text/plain')` will return a String. We've accounted for both possibilities in `navigator.clipboard.write` just in case the polyfill isn't applied (e.g. Chrome adds promises but keeps their buggy `getType()` behaviour).

This means `await navigator.clipboard.write(await navigator.clipboard.read())` should be a no-op (as long as the user only has plain-text data in their clipboard) which feels like a sensible thing to do.

### Customer Support Data

Here are some tables representing the % of browsers we see that support (or don't support) these APIs.

#### [Clipboard Write](https://developer.mozilla.org/docs/Web/API/Clipboard/write)
| Browser                 | Supported Since | Latest Version | % Supported | % Unsupported |
| :---------------------- | --------------: | -------------: | ----------: | ------------: |
| chrome                  |              66 |            101 |      73.565 |         0.007 |
| edge                    |              79 |            100 |       6.691 |         0.001 |
| firefox                 |   Not Supported |            100 |       0.000 |        12.669 |
| firefox_android         |   Not Supported |            100 |       0.000 |         0.128 |
| ie                      |   Not Supported |             11 |       0.000 |         0.078 |
| opera                   |              63 |             86 |       1.267 |         0.000 |
| safari                  |            13.1 |           15.4 |       4.630 |         0.013 |
| safari_ios              |            13.4 |           15.4 |       0.505 |         0.006 |
| samsunginternet_android |            12.0 |           16.0 |       0.020 |         0.000 |
| webview_android         |              66 |            101 |       0.002 |         0.008 |
| **Total:**              |                 |                |   **86.68** |     **12.91** 

 (Note: "Other" browsers make up 0.408%)

### CODE REVIEW

**Comment by pablonete** [comment]
This is great, thank you @keithamus for putting this together 🎉 
I've read through it and I have no concerns, just a couple of questions:
1. Is this going to provide Firefox support too?
2. Will this be added to `index.ts`, so the `apply` method brings it to dotcom? I'm just trying to clarify the road ahead: once we merge this and upgrade this lib in dotcom, we'd got https://github.com/github/repos/issues/1073 fixed 🙏

File: src/navigator-clipboard.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export async function clipboardWrite(data: ClipboardItems) {
+  if (data.length === 0) return
+  if (data.length > 1) throw new Error('Support for multiple ClipboardItems is not implemented.')
```
Comment (theinterned) [inline src/navigator-clipboard.ts:3]: I'm not sure what means. Would multiple ClipBoard items for example mean that you have multiple representations of the same data (for example `text/plain` and `text/html` ...)?

Will this be a common occurrence? Is it easily avoidable in downstream code?

**Review by theinterned — COMMENTED** [review]
This is super cool, just left my one drive-by question

File: src/navigator-clipboard.ts (hunk 1 above)
Comment (keithamus) [reply src/navigator-clipboard.ts:3]: This isn't defined by us, but is [per the spec](https://w3c.github.io/clipboard-apis/#typedefdef-clipboarditems). `data` must be an array of ClipboardItems.

Having said that we're doing the wrong thing here. In the steps (https://w3c.github.io/clipboard-apis/#dom-clipboard-write) it says:

> If data’s [size](https://infra.spec.whatwg.org/#list-size) is greater than 1, and the current operating system does not support multiple native clipboard items on the [system clipboard](https://w3c.github.io/clipboard-apis/#system-clipboard), then add data[0] to dataList, else, set dataList to data.

So we should just get the first item of the array instead.

**Review by pablonete — DISMISSED** [review]

**Review by koddsson — DISMISSED** [review]

File: docs/index.html [hunk 2]
```diff
@@ -153,6 +154,8 @@ <h1>GitHub Feature Support Table</h1>
               <span class="unsupported">!</span>Required feature, not available in this browser.
             </p><p>
               <span class="polyfilled">*</span>Not avaible in this browser, but polyfilled using this library.
+            </p><p>
+              <span class="buggy"><small>†</small></span>Required feature, buy polyfilled to smooth over bugs in this browser.
```
Comment (stof) [inline docs/index.html:158]: typo. `buy` vs `but`

**Comment by pablonete** [comment]
Tested today on Firefox and confirmed that this fixes the Copy Remote functionality in the blob page for Firefox, see details on https://github.com/github/repos/issues/1073#issuecomment-1226971804.

**Review by koddsson — APPROVED** [review]

**Review by JoseInTheArena — APPROVED** [review]
lgtm 🚀

**Pull request merged**